python/lsst/ts/externalscripts/make_love_stress_tests.py

```diff
@@ -0,0 +1,324 @@
+        - LOVE host location
+        - Number of clients
+        - Number of messages to store
+        - CSCs
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Configuration with several attributes, defined in `get_schema`
+
+        Notes
+        -----
+        Saves the results on several attributes:
+
+        * config    : `types.SimpleNamespace`, same as config param
+        * remotes   : a dict, with each item as
+            CSC_name[:index]: `lsst.ts.salobj.Remote`
+
+        Constructing a `salobj.Remote` is slow (DM-17904), so configuration
+        may take a 10s or 100s of seconds per CSC.
+        """
+        self.log.info("Configure started")
+
+        # set configurations
+        self.config = config
+
+        # get credentials
+        self.username = "user"
+        self.password = os.environ.get("USER_USER_PASS")
+        if self.password is None:
+            raise RuntimeError(
+                "Configuration failed: environment variable USER_USER_PASS not defined"
+            )
+
+        # construct remotes
+        remotes = dict()
+        for name_index in config.data:
+            name, index = salobj.name_to_name_index(name_index)
+            self.log.debug(f"Create remote {name}:{index}")
+            if name_index not in remotes:
+                remote = salobj.Remote(domain=self.domain, name=name, index=index)
+                remotes[name_index] = remote
+        self.remotes = remotes
+
+    async def run(self):
+        """Run script."""
+
+        self.log.info(f"Waiting for {len(self.remotes)} remotes to be ready")
+        await asyncio.gather(*[remote.start_task for remote in self.remotes.values()])
+
+        # Enable all CSCs
+        for remote in self.remotes.values():
+            await salobj.set_summary_state(remote=remote, state=salobj.State.ENABLED)
+
+        event_streams = dict()
+        telemetry_streams = dict()
+        for name_index in self.remotes:
+            name, index = salobj.name_to_name_index(name_index)
+            salinfo = salobj.SalInfo(self.domain, name)
+            try:
+                event_streams[name_index] = salinfo.event_names
+                telemetry_streams[name_index] = salinfo.telemetry_names
+            finally:
+                await salinfo.close()
+
+        # Create clients and listen to ws messages
+        self.log.info(
+            f"Waiting for {self.config.number_of_clients} Manager Clients to be ready"
+        )
+        loop = asyncio.get_event_loop()
+        for i in range(self.config.number_of_clients):
+            self.clients.append(
+                ManagerClient(
+                    self.config.location,
+                    self.username,
+                    self.password,
+                    event_streams,
+                    telemetry_streams,
+                )
+            )
+
+        for client in self.clients:
+            loop.create_task(client.start_ws_client())
```

Comment:
As I mentioned above, I think you can do `asyncio.create_task(...)` instead.

Reply:
Also this code will not work reliably. You must keep a reference to each task you create until the task finishes. Otherwise you risk it being garbage-collected and canceled before it is done.

Two options are to accumulate the tasks in a separate list, or inside the clients. I usually do the latter (if I wrote the class), because it makes cancelling much easier: the close/cancel/stop method of the object can cancel all such internal tasks. This might do:
```
class Client:
    def __init__(self, ...):
        self.start_task = utils.make_done_future()

    def create_start_task(self):
        self.start_task = asyncio.create_task(self.start_ws_client())
```
The name is a bit ugly, but indicates that the method creates the task without waiting for it to finish. (A name like "start" implies that tit is async and waits until the task finishes).